api/krusty/helmchartinflationgenerator_test.go

```diff
@@ -21,21 +21,46 @@ kind: Secret
 metadata:
   labels:
     app: test-minecraft
-    chart: minecraft-3.1.3
+    app.kubernetes.io/instance: test-minecraft
+    app.kubernetes.io/name: minecraft
+    app.kubernetes.io/version: 4.26.4
+    chart: minecraft-4.26.4
     heritage: Helm
     release: test
-  name: test-minecraft
+  name: test-minecraft-rcon
+  namespace: default
+type: Opaque
+---
+apiVersion: v1
+data:
+  cf-api-key: Q0hBTkdFTUUh
+kind: Secret
+metadata:
+  labels:
+    app: test-minecraft
+    app.kubernetes.io/instance: test-minecraft
+    app.kubernetes.io/name: minecraft
+    app.kubernetes.io/version: 4.26.4
+    chart: minecraft-4.26.4
+    heritage: Helm
+    release: test
+  name: test-minecraft-curseforge
+  namespace: default
```

Comment:
This is a genuine question: why were major changes made? Was it necessary to update helm charts when running tests?

If updating helm charts isn't absolutely required for test execution, I'd appreciate if these changes could be separated from the main update.

Reply:
The helm chart updates were necessary to properly test the namespace propagation fix.

The older chart versions (some 4+ years old) had a bug where `{{ .Release.Namespace }}` wasn't correctly utilized. Newer versions fixed this, but our tests were still using the outdated versions, making it impossible to validate that namespaces propagate correctly from Kustomize to Helm.

Updating the charts ensures:
1. Tests reflect real-world usage (not 4-year-old charts)
2. We can properly validate the namespace propagation fix
3. Consistency across tests instead of a confusing mix of versions